mkosi/__init__.py

```diff
@@ -4718,6 +4718,9 @@ def _link_output(
 
 
 def link_output(args: MkosiArgs, root: Path, artifact: Optional[BinaryIO]) -> None:
+    if not root.exists():
+        return
```

Comment:
This should only be done in case the output format is directory/subvolume. Otherwise, we should check if `artifact` is `None` and skip if that's the case.